plugins/VstBase/RemoteVstPlugin.cpp

```diff
@@ -1365,36 +1371,47 @@ void RemoteVstPlugin::rotateProgram( int offset )
 
 void RemoteVstPlugin::getProgramNames()
 {
-	char presName[1024+256*30];
-	char curProgName[30];
-	if (isInitialized() == false) return;
-	bool progNameIndexed = ( pluginDispatch( 29, 0, -1, curProgName ) == 1 );
-
-	if (m_plugin->numPrograms > 1) {
-		if (progNameIndexed) {
-			for (int i = 0; i< (m_plugin->numPrograms >= 256?256:m_plugin->numPrograms); i++)
+	char presName[1024 + 256 * 30] = {};
+	constexpr auto maxlen = sizeof(presName) - 1;
+	char curProgName[30] = {};
+	if (!isInitialized()) { return; }
+	const bool progNameIndexed = pluginDispatch(29, 0, -1, curProgName) == 1;
+
+	if (m_plugin->numPrograms > 1)
+	{
+		const auto maxPrograms = std::min(m_plugin->numPrograms, 256);
+		if (progNameIndexed)
+		{
+			for (int i = 0; i < maxPrograms; i++)
 			{
-				pluginDispatch( 29, i, -1, curProgName );
-				if (i == 0) 	sprintf( presName, "%s", curProgName );
-				else		sprintf( presName + strlen(presName), "|%s", curProgName );
+				pluginDispatch(29, i, -1, curProgName);
+				if (i == 0) { snprintf(presName, maxlen, "%s", curProgName); }
+				else
+				{
+					const auto len = strlen(presName);
+					snprintf(presName + len, maxlen - len, "|%s", curProgName);
+				}
 			}
 		}
 		else
 		{
-			int currProgram = pluginDispatch( effGetProgram );
-			for (int i = 0; i< (m_plugin->numPrograms >= 256?256:m_plugin->numPrograms); i++)
+			const int currProgram = pluginDispatch(effGetProgram);
+			for (int i = 0; i < maxPrograms; i++)
 			{
-				pluginDispatch( effSetProgram, 0, i );
-				if (i == 0) 	sprintf( presName, "%s", programName() );
-				else		sprintf( presName + strlen(presName), "|%s", programName() );
+				pluginDispatch(effSetProgram, 0, i);
+				if (i == 0) { snprintf(presName, maxlen, "%s", programName()); }
+				else
+				{
+					const auto len = strlen(presName);
+					const auto remaining = std::min(len, maxlen);
```

Comment:
Should this be `maxlen - len` like in the `if (progNameIndexed)` branch of the if-statement?

Reply:
Yeah, probably. I'll change that :+1: